libffi-sys-rs/build/not_msvc.rs

```diff
@@ -17,9 +17,16 @@ pub fn build_and_link() {
             e
         );
     }
+
+    // On Linux, don't preserve the attributes of the source directory.
+    let mut command = Command::new("cp");
+    let host = std::env::var("HOST").unwrap();
+    if host.ends_with("-linux-gnu") {
+        command.arg("--no-preserve=mode,ownership");
+    };
```

Comment:
Can't this be done using `#[cfg(target_os = "linux")]` / `cfg!(target_os = "linux")` instead?

Reply:
Oh, just as I write that I realise that might not work when cross-compiling? 

In that case, this change would only mean the option is applied when compiling using a GNU environment. If somebody compiles for musl, I think this check fails. Perhaps it's better to just see if "linux" is present in `HOST`?

Reply:
Alternatively: we can check the output of `cp --help` to see if it supports `--no-preserve`. That way it doesn't matter what the platform used is.